Terminal.Gui/View/ViewDrawing.cs

```diff
@@ -503,5 +525,284 @@ public virtual void OnDrawContentComplete (Rect contentArea)
 			DrawContentComplete?.Invoke (this, new DrawEventArgs (contentArea));
 		}
 
+		/// <summary>
+		/// Draws a rectangular frame. The frame will be merged (auto-joined) with any other lines drawn by this View 
+		/// if <paramref name="mergeWithLineCanvas"/> is true, otherwise will be rendered immediately.
+		/// </summary>
+		/// <param name="rect">The view relative location and size of the frame.</param>
+		/// <param name="lineStyle">The line style.</param>
+		/// <param name="attribute">The colors to be used.</param>
+		/// <param name="mergeWithLineCanvas">When drawing the frame, allow it to integrate (join) to other frames in other controls.
+		/// Or false to simply draw the rect exactly with no side effects.</param>
+		public void DrawFrame (Rect rect, LineStyle lineStyle, Attribute? attribute = null, bool mergeWithLineCanvas = true)
+		{
+			LineCanvas lc;
+			if (mergeWithLineCanvas) {
+				lc = new LineCanvas ();
+			} else {
+				lc = LineCanvas;
+			}
+			var vts = ViewToScreen (rect);
+			lc.AddLine (new Point (vts.X, vts.Y), vts.Width,
+				Orientation.Horizontal, lineStyle, attribute);
+			lc.AddLine (new Point (vts.Right - 1, vts.Y), vts.Height,
+				Orientation.Vertical, lineStyle, attribute);
+			lc.AddLine (new Point (vts.Right - 1, vts.Bottom - 1), -vts.Width,
+				Orientation.Horizontal, lineStyle, attribute);
+			lc.AddLine (new Point (vts.X, vts.Bottom - 1), -vts.Height,
+				Orientation.Vertical, lineStyle, attribute);
+
+			if (mergeWithLineCanvas) {
+				LineCanvas.Merge (lc);
+			} else {
+				OnRenderLineCanvas ();
+			}
+		}
+
+		/// <summary>
+		/// Draws an incomplete frame. The frame will be merged (auto-joined) with any other lines drawn by this View 
+		/// if <paramref name="mergeWithLineCanvas"/> is true, otherwise will be rendered immediately.
+		/// The frame is always drawn clockwise. For <see cref="Side.Top"/> and <see cref="Side.Right"/> the end position must
+		/// be greater or equal to the start and for <see cref="Side.Left"/> and <see cref="Side.Bottom"/> the end position must
+		/// be less or equal to the start.
+		/// </summary>
+		/// <param name="startPos">The view relative of the start and side position.</param>
+		/// <param name="endPos">The view relative of the end and side position.</param>
+		/// <param name="rect">The view relative location and size of the frame.</param>
+		/// <param name="lineStyle">The line style.</param>
+		/// <param name="attribute">The colors to be used.</param>
+		/// <param name="mergeWithLineCanvas">When drawing the frame, allow it to integrate (join) to other frames in other controls.
+		/// Or false to simply draw the rect exactly with no side effects.</param>
+		public void DrawIncompleteFrame ((int start, Side side) startPos, (int end, Side side) endPos, Rect rect, LineStyle lineStyle, Attribute? attribute = null, bool mergeWithLineCanvas = true)
```

Comment:
@BDisp I do not understand what all this complex logic is for. Can you please describe the problem you are trying to solve with this?

Reply:
It's a helper method to draw incomplete frames based on start and end. I'm tying to implement it into the `TabView`. If you see the unit tests you'll understand the logic.